docs/errors.md

```diff
@@ -0,0 +1,67 @@
+# LNDK Error Handling
+
+LNDK returns structured error information through gRPC status codes and error details to help clients handle different failure scenarios.
+
+## Error Structure
+
+LNDK errors have three components:
+1. **gRPC Status Code**: Standard codes like `INVALID_ARGUMENT`, `INTERNAL`, `UNAVAILABLE`
+2. **Error Code**: LNDK-specific identifier (e.g., `PARSE_OFFER_FAILURE`, `INVALID_AMOUNT`)
+3. **Human Message**: Descriptive error message
+
+## Error Categories
+
+LNDK returns three main error types:
+- **OfferError**: Offer parsing, validation, and payment processing errors
+- **LndError**: LND connectivity and configuration errors  
+- **AuthError**: Authentication and authorization errors
+
+## Using grpcurl
+
+`grpcurl` is a command-line tool for testing gRPC services. Here's how to use it with LNDK:
+
+
+### Example
+
+Test with an invalid offer to see error details:
+
+```bash
+grpcurl -insecure \
+  -H "macaroon: $(xxd -ps -u -c 1000 '~/.lnd/data/chain/bitcoin/regtest/admin.macaroon')" \
+  -import-path proto \
+  -import-path vendor \
+  -proto proto/lndkrpc.proto \
+  -d '{"offer": "", "amount": 10000}' \
+  127.0.0.1:7000 lndkrpc.Offers/GetInvoice
+```
+
+Response:
+```
+ERROR:
+  Code: InvalidArgument
+  Message: The provided offer was invalid. Please provide a valid offer in bech32 format, i.e. starting with 'lno'. Error: Bech32(Parse(Char(MissingSeparator)))
+  Details:
+  1)    {
+          "@type": "type.googleapis.com/google.rpc.ErrorInfo",
+          "domain": "lndk",
+          "reason": "PARSE_OFFER_FAILURE"
+        }
+```
+
+## Client Requirements
+
+To properly see LNDK's structured error details in your client, you need:
+
+### 1. gRPC Error Details Support
+Your client must support gRPC's rich error details mechanism (`google.rpc.ErrorInfo`). This requires:
+- Import `google/rpc/error_details.proto` 
```

Comment:
nit: I'd add where is located in our structure and link it to where is the original source